planner/core/preprocess_test.go

```diff
@@ -194,6 +194,9 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"select * from ( select 1 ) a, (select 2) b;", true, nil},
 		{"select * from (select * from ( select 1 ) a join (select 2) b) b join (select 3) a;", false, nil},
 		{"select * from (select 1 ) a , (select 2) b, (select * from (select 3) a join (select 4) b) c;", false, nil},
+
+		//issue 7833
+		{"drop view if exists v;", true, nil},
```

Comment:
Better to add this test in `TestColumnPruning`.

Reply:
Well, I've found that this test case can be added here（planner/core/preprocess_test.go）, and even without fixing the bug, I'm confused, maybe I don't fully understand the tidb test.